templates/python/custom-copilot-rag-azure-ai-search/src/AzureAISearchDataSource.py

```diff
@@ -0,0 +1,86 @@
+from dataclasses import dataclass
+from typing import Optional, List
+from azure.search.documents.indexes.models import _edm as EDM
+from azure.search.documents.models import VectorQuery, VectorizedQuery
+from teams.ai.embeddings import OpenAIEmbeddings, AzureOpenAIEmbeddings
+from teams.state.memory import Memory
+from teams.state.state import TurnContext
+from teams.ai.tokenizers import Tokenizer
+
+from src.indexers.data import get_embedding_vector
+
+@dataclass
+class Doc:
+    docId: Optional[str] = None
+    docTitle: Optional[str] = None
+    description: Optional[str] = None
+    descriptionVector: Optional[List[float]] = None
+
+@dataclass
+class AzureAISearchDataSourceOptions:
+    name: str
+    indexName: str
+    azureAISearchApiKey: str
+    azureAISearchEndpoint: str
+
+from abc import ABC, abstractmethod
+from azure.core.credentials import AzureKeyCredential
+from azure.search.documents import SearchClient
+import json
+
+class DataSource(ABC):
```

Comment:
Does Teams-AI exposes a data source interface? We may use the interface from Teams-AI if possible.

Reply:
will check